File: src/week1/class2/assignment/runningSumOf1dArray/RunningSumOf1dArray.java
package week1.class2.assignment.runningSumOf1dArray;

import java.util.Arrays;


public class RunningSumOf1dArray {

    public static void main(String[] args) {
        int[] inputArray = {1, 2, 3, 4};
        String result = Arrays.toString(sum(inputArray));
        System.out.println(result);
    }

    static int[] sum(int[] input) {
        int[] resultantArray = new int[input.length];

        for (int i = 0; i < input.length; i++) {
            int runningSum = 0;
            for (int j = 0; j <= i; j++) {
                runningSum += input[j];
            }
            resultantArray[i] = runningSum;
        }

        return resultantArray;
    }
}

File: src/week1/class2/assignment/RunningSumOf1DArray.kt
package week1.class2.assignment

import kotlin.math.sign

/*
* Given an array nums. We define a running sum of an array as runningSum[i] = sum(nums[0]…nums[i]).

Return the running sum of nums.



Example 1:

Input: nums = [1,2,3,4]
Output: [1,3,6,10]
Explanation: Running sum is obtained as follows: [1, 1+2, 1+2+3, 1+2+3+4].
Example 2:

Input: nums = [1,1,1,1,1]
Output: [1,2,3,4,5]
Explanation: Running sum is obtained as follows: [1, 1+1, 1+1+1, 1+1+1+1, 1+1+1+1+1].
Example 3:

Input: nums = [3,1,2,10,1]
Output: [3,4,6,16,17]


Constraints:

1 <= nums.length <= 1000
-10^6 <= nums[i] <= 10^6
*
* */

fun main() {
    val input = intArrayOf(1, 2, 3, 4)
    val result = RunningSumOf1DArray().runningSum(input)
    println(result.toList())
}

class RunningSumOf1DArray {
    fun runningSum(nums: IntArray): IntArray {
        for (index in 1..<nums.size) {
            nums[index] = nums[index - 1] + nums[index]
        }
        return nums;
    }
}